pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -510,42 +505,25 @@ private void processStreamEvents(MessageBatch messagesAndOffsets, long idlePipeS
           .decode(messagesAndOffsets.getMessageAtIndex(index), messagesAndOffsets.getMessageOffsetAtIndex(index),
               messagesAndOffsets.getMessageLengthAtIndex(index), reuse);
       if (decodedRow != null) {
-        List<GenericRow> transformedRows = new ArrayList<>();
+        TransformPipeline.Result result = new TransformPipeline.Result();
         try {
-          if (_complexTypeTransformer != null) {
-            // TODO: consolidate complex type transformer into composite type transformer
-            decodedRow = _complexTypeTransformer.transform(decodedRow);
-          }
-          Collection<GenericRow> rows = (Collection<GenericRow>) decodedRow.getValue(GenericRow.MULTIPLE_RECORDS_KEY);
-          if (rows != null) {
-            for (GenericRow row : rows) {
-              GenericRow transformedRow = _recordTransformer.transform(row);
-              if (transformedRow != null && IngestionUtils.shouldIngestRow(row)) {
```

Comment:
I noticed that here `shouldIngestRow` is tested against row **before** `_recordTransformer.transform(row);`, however below when it is not multiple record, we test against the row **after** the transform:  `IngestionUtils.shouldIngestRow(transformedRow)` . Is this expected or we should correct that?

Reply:
Good catch. We should test against the transformed row (after the transformation). It works now because they are always the same reference per the current implementation of transformer

Reply:
Changed to test on transforedRow now